backend/src/utils/db/recurring_charges.py

```
@@ -0,0 +1,504 @@
+    
+    # First, verify the pattern exists and belongs to the user
+    existing_pattern = get_pattern_by_id_from_db(pattern_id, user_id)
+    if not existing_pattern:
+        raise NotFound(f"Pattern {str(pattern_id)} not found")
+    
+    # Extract only the fields that were set (not None) from the DTO
+    updates = pattern_update.model_dump(by_alias=True, exclude_unset=True, exclude_none=True)
+    
+    if not updates:
+        # No updates provided, return existing pattern
+        logger.debug(f"DB: No updates provided for pattern {str(pattern_id)}")
```

Ruff (0.14.3)

Use explicit conversion flag

Replace with conversion flag

(RUF010)